pkgs/desktops/cinnamon/xapps/default.nix

```diff
@@ -0,0 +1,77 @@
+{ fetchFromGitHub
+, fetchpatch
+, glib
+, gobject-introspection
+, gtk3
+, libgnomekbd
+, libxklavier
+, meson
+, ninja
+, pkgconfig
+, python3
+, python3Packages
+, stdenv
+, vala
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xapps";
+  version = "1.6.4";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "08c8j7wv56zf8mfgjfhd0wc1bbrw0dj1lmlawvsxwfajnnyjsm0d";
+  };
+
+  NIX_CFLAGS_COMPILE = [ "-I${glib.dev}/include/gio-unix-2.0" ]; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
+
+  patches = [
+    (fetchpatch { # details see https://github.com/linuxmint/xapps/pull/65
+      url = "https://github.com/linuxmint/xapps/pull/65.diff";
+      sha256 = "1imcf6nfa1ky0nkirfim3fa7bx72w3k5a5p66bigcm14n0lqli7w";
+    })
+  ];
+
+  buildInputs = [
+    glib
+    gtk3
+    libgnomekbd
+    libxklavier
+    (python3.withPackages(ps: with ps; [ pygobject3 ]))
+  ];
+
+  nativeBuildInputs = [
+    meson
+    gobject-introspection
+    ninja
+    wrapGAppsHook
+    pkgconfig
+    vala
+
+    python3Packages.pygobject3 # otherwise pkgconfig can't find it
```

Comment:
This should go to `buildInputs` as it is runtime dependency still. (Even though it is included into the package through different means.)